Framework/Core/README.md

```diff
@@ -650,20 +650,42 @@ will then be made available at workflow creation time via the `ConfigContext`
 passed to the `defineDataProcessing` function, using the `ConfigContext::options()`
 getter.
 
-# Forseen features
+## Completion policies (WIP)
+
+By default the data processing of a given record happens when all the fields of
+that record are present. So if your Data Processor declares it will consume
+`TRACKS` and `CLUSTERS`, for any given time interval both need to be produced
+by some other data processor before the computation declared in yours can happen.
+
+Sometimes it's however desirable to customise such a behavior, so that some action
+on the record can happen even if it's not complete. For example you might want
+to start computing some quantity as a given message arrives and then complete the
+computation once the record is complete. This is done by specifying by customising 
+the data processing CompletionPolicy. This can be done using the usual **Customization
+mechanism** where a:
+
+```C++
+void customize(std::vector<CompletionPolicy> &policies);
+```
```

Comment:
One more thought: Isn't the Completion policy something specific to a DataProcessor? So I feel it might rather belong into a DataProcessorSpec instead of being specified in the global workflow?

Reply:
I thought about it and could not make up my mind.

 In principle a `DeviceSpec` should give you access to more information compared to a `DataProcessorSpec`. For example a `DeviceSpec` knows who is actually sending you the data (i.e. you get an `InputRoute`, rather than an `InputSpec`).

Do you have any information you would like to use from the `DataProcessorSpec` which is not mirrored in the `DeviceSpec`?

Reply:
`DataProcessorLabel`'s could find its place in `DeviceSpec`, also for the usage of `ChannelConfigurationPolicy`'ies

Reply:
Agreed. I will try to expose those or even the whole DataProcessorSpec.